kong/plugins/ai-proxy/drivers/anthropic.lua

```diff
@@ -0,0 +1,250 @@
+end
+
+
+local transformers_to = {
+  ["llm/v1/chat"] = function(request_table, model)
+    local prompt = {}
+    local err
+
+    prompt.prompt, err = to_claude_prompt(request_table)
+    if err then return nil, nil, err end
+    
+    prompt.temperature = (model.options and model.options.temperature) or nil
+    prompt.max_tokens_to_sample = (model.options and model.options.max_tokens) or nil
+    prompt.model = model.name
+
+    return prompt, "application/json", nil
+  end,
+
+  ["llm/v1/completions"] = function(request_table, model)
+    local prompt = {}
+    local err
+
+    prompt.prompt, err = to_claude_prompt(request_table)
+    if err then return nil, nil, err end
+    
+    prompt.temperature = (model.options and model.options.temperature) or nil
+    prompt.max_tokens_to_sample = (model.options and model.options.max_tokens) or nil
+    prompt.model = model.name
+
+    return prompt, "application/json", nil
+  end,
+}
+
+local transformers_from = {
+  ["llm/v1/chat"] = function(response_string)
+    local response_table, err = cjson.decode(response_string)
+    if err then
+      return nil, "failed to decode cohere response"
+    end
+
+    if response_table.completion then
+      local res = {
+        choices = {
+          {
+            index = 0,
+            message = {
+              role = "assistant",
+              content = response_table.completion,
+            },
+            finish_reason = response_table.stop_reason,
+          },
+        },
+        model = response_table.model,
+        object = "chat.completion",
+      }
+        
+      return cjson.encode(res)
+    else
+      -- it's probably an error block, return generic error
+      return nil, "'completion' not in anthropic://llm/v1/chat response"
+    end
+  end,
+
+  ["llm/v1/completions"] = function(response_string)
+    local response_table, err = cjson.decode(response_string)
+    if err then
+      return nil, "failed to decode cohere response"
+    end
+
+    if response_table.completion then
+      local res = {
+        choices = {
+          {
+            index = 0,
+            text = response_table.completion,
+            finish_reason = response_table.stop_reason,
+          },
+        },
+        model = response_table.model,
+        object = "text_completion",
+      }
+
+      return cjson.encode(res)
+    else
+      -- it's probably an error block, return generic error
+      return nil, "'completion' not in anthropic://llm/v1/chat response"
+    end
+  end,
+}
+
+function _M.from_format(response_string, model_info, route_type)
+  -- MUST return a string, to set as the response body
+  ngx.log(ngx.DEBUG, "converting from ", model_info.provider, "://", route_type, " type to kong")
+
+  if not transformers_from[route_type] then
+    return nil, fmt("no transformer available from format %s://%s", model_info.provider, route_type)
+  end
+  
+  local ok, response_string, err = pcall(transformers_from[route_type], response_string)
```

Comment:
Suggestion: stashing the table lookup to a local var...

```suggestion
  local transform = transformers_from[route_type]
  if not transform then
    return nil, fmt("no transformer available from format %s://%s", model_info.provider, route_type)
  end
  
  local ok, response_string, err = pcall(transform, response_string)
```

Rationale: it's functionally equivalent to what you have now, but it makes it a little safer from accidentally becoming unsafe via refactoring later on. Imagine if some time in the future we want to update the value of `route_type` for some special case...

```lua
  if not transformers_from[route_type] then
    return nil, fmt("no transformer available from format %s://%s", model_info.provider, route_type)
  end

  if special_circumstance() then
    -- update the route type for whatever reason
    route_type = "my-new-route-type"
  end

  -- oops, now the behavior of this code has changed
  local ok, response_string, err = pcall(transformers_from[route_type], response_string)
```


Alternatively, if `transformers_from[route_type]` is _always_ expected to be valid at runtime (e.g. `route_type` originates from an enum or something), then using `assert()` as a last ditch sanity check is also fine and simplifies the code:

```lua
local transform = assert(transformers_from[route_type], "no transformer for this route type")
```